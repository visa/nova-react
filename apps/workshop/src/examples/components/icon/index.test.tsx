/**
 *              Â© 2025 Visa
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 **/
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';

import { CustomColorIcon } from './custom-color-icon';
import { DefaultIcon } from './default-icon';
import { VisaIcon } from './visa-icon';
import { TinyResolution } from './tiny-resolution';
import { LowResolution } from './low-resolution';
import { HighResolution } from './high-resolution';
import { GenericIcon } from './generic-icon';
import metaData from './meta.json';
import { RtlIcons } from './rtl-icon';

const examples = [
  { Component: DefaultIcon, title: metaData['default-icon'].title },
  { Component: VisaIcon, title: metaData['visa-icon'].title },
  { Component: GenericIcon, title: metaData['generic-icon'].title },
  { Component: TinyResolution, title: metaData['tiny-resolution'].title },
  { Component: LowResolution, title: metaData['low-resolution'].title },
  { Component: HighResolution, title: metaData['high-resolution'].title },
  { Component: RtlIcons, title: metaData['rtl-icon'].title },
  { Component: CustomColorIcon, title: metaData['custom-color-icon'].title },
];

describe('Icon examples', () => {
  examples.forEach(({ Component, title }) => {
    it(`${title} should render correctly`, async () => {
      const { container } = render(<Component />);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
      expect(container).toMatchSnapshot();
    });
  });
});
