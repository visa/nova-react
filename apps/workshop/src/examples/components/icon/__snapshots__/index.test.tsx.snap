// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon examples Default should render correctly 1`] = `
<div>
  <svg
    aria-hidden="false"
    aria-label="global"
    class="v-icon v-icon-visa v-icon-low v-icon-global"
    focusable="false"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      class="v-icon-secondary"
      d="M13.7239 23C13.3068 23 12.9 22.9798 12.483 22.9395C9.37058 22.6167 6.58367 21.1137 4.61045 18.7029C1.04036 14.3251 1.14207 8.06098 4.85456 3.80422C5.3021 3.28978 5.80049 2.81568 6.33956 2.38193C6.5735 2.19028 7.45839 1.58505 7.72284 1.42366C7.9873 1.27235 8.24158 1.13113 8.5162 1L9.42144 2.80559C9.19767 2.91655 8.98408 3.02751 8.77048 3.15864C8.5874 3.25951 7.78387 3.82439 7.62113 3.94544C7.1736 4.30857 6.75658 4.70197 6.39042 5.12563C3.31871 8.63595 3.23734 13.8106 6.19716 17.4218C7.82456 19.409 10.1334 20.6497 12.6966 20.9221C15.2699 21.1843 17.7822 20.448 19.7859 18.834C20.386 18.3599 20.9149 17.8051 21.3828 17.21L23 18.4406C22.4406 19.1669 21.7896 19.8226 21.0675 20.4076C18.962 22.1022 16.3887 23 13.7239 23Z"
    />
    <path
      class="v-icon-primary"
      d="M19.3284 6.77922C18.0715 5.21869 16.2775 4.24589 14.2807 4.04322C12.2941 3.83042 10.3378 4.40802 8.77688 5.67468C7.21594 6.93122 6.24289 8.72481 6.04017 10.7211C5.83745 12.7173 6.40507 14.6629 7.67206 16.2235C8.92892 17.784 10.723 18.7568 12.7198 18.9595C12.9833 18.9899 13.257 19 13.5205 19C15.2335 19 16.8755 18.4224 18.2236 17.328C19.7845 16.0715 20.7576 14.2779 20.9603 12.2816C21.163 10.2955 20.5853 8.33975 19.3284 6.77922ZM14.0678 6.05975C15.0105 6.16108 15.872 6.50562 16.6221 7.04268L14.9902 8.67415C13.9766 7.72162 12.9022 6.95148 11.919 6.49548C11.8075 6.44482 11.7163 6.41442 11.6149 6.37388C12.213 6.16108 12.8313 6.03948 13.48 6.03948C13.6827 6.02935 13.8753 6.03948 14.0678 6.05975ZM16.3687 14.2576C15.9633 14.2373 14.9598 13.8523 13.6827 12.8389L14.9497 11.5723C15.9633 12.8491 16.3484 13.8523 16.3687 14.2576ZM12.1927 11.4608C10.6824 9.83948 10.1351 8.50188 10.1148 8.02561C10.6013 8.04588 11.9393 8.59308 13.5509 10.1029L12.1927 11.4608ZM8.05723 10.9237C8.10791 10.4171 8.25995 9.93068 8.44239 9.46455C8.49307 9.58615 8.52348 9.69761 8.5843 9.81921C9.04041 10.8021 9.82088 11.8763 10.7635 12.8896L9.03028 14.6224C8.28022 13.5381 7.91532 12.2512 8.05723 10.9237ZM12.9225 16.9429C12.0305 16.8517 11.2095 16.5275 10.4797 16.0411L12.2332 14.288C13.0137 14.9264 13.8144 15.4635 14.5544 15.808C15.1828 16.1019 15.6997 16.2336 16.1457 16.2741C15.1727 16.8112 14.0577 17.0544 12.9225 16.9429ZM18.9432 12.0688C18.8723 12.7477 18.6696 13.3861 18.3655 13.984C18.2337 12.8187 17.433 11.4203 16.389 10.1435L18.0411 8.49175C18.7405 9.54561 19.075 10.792 18.9432 12.0688Z"
    />
  </svg>
</div>
`;

exports[`Icon examples Generic should render correctly 1`] = `
<div>
  <svg
    aria-hidden="false"
    aria-label="info"
    class="v-icon v-icon-generic v-icon-low v-icon-information"
    focusable="false"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      class="v-icon-primary"
      clip-rule="evenodd"
      d="M22 12C22 6.47728 17.5227 2 12 2C6.47728 2 2 6.47728 2 12C2 17.5227 6.47728 22 12 22C17.5227 22 22 17.5227 22 12ZM24 12C24 18.6273 18.6273 24 12 24C5.37272 24 0 18.6273 0 12C0 5.37272 5.37272 0 12 0C18.6273 0 24 5.37272 24 12ZM10.0898 17.9795H11.5898L13.2598 9.26953H11.7698L10.0898 17.9795ZM12.9998 5.83008C12.4475 5.83008 11.9998 6.27779 11.9998 6.83008C11.9773 7.07168 12.0578 7.31157 12.2216 7.4906C12.3854 7.66963 12.6172 7.77113 12.8598 7.77008C13.4121 7.77008 13.8598 7.32236 13.8598 6.77008C13.8793 6.5292 13.7977 6.29109 13.6345 6.11279C13.4714 5.93449 13.2415 5.83207 12.9998 5.83008Z"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;

exports[`Icon examples High resolution should render correctly 1`] = `
<div>
  <svg
    aria-hidden="false"
    aria-label="info"
    class="v-icon v-icon-visa v-icon-high v-icon-information"
    focusable="false"
    height="48"
    viewBox="0 0 48 48"
    width="48"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      class="v-icon-secondary"
      d="M24 8C15.1777 8 8 15.1777 8 24C8 32.8223 15.1777 40 24 40C32.8223 40 40 32.8223 40 24C40 15.1777 32.8223 8 24 8ZM24 38C16.2803 38 10 31.7197 10 24C10 16.2803 16.2803 10 24 10C31.7197 10 38 16.2803 38 24C38 31.7197 31.7197 38 24 38Z"
    />
    <path
      class="v-icon-primary"
      clip-rule="evenodd"
      d="M4 24C4 12.9722 12.9722 4 24 4C35.0283 4 44 12.9722 44 24C44 35.0283 35.0283 44 24 44C12.9722 44 4 35.0283 4 24ZM6 24C6 33.9248 14.0747 42 24 42C33.9248 42 42 33.9248 42 24C42 14.0747 33.9248 6 24 6C14.0747 6 6 14.0747 6 24ZM24 20C25.6543 20 27 18.6543 27 17C27 15.3457 25.6543 14 24 14C22.3457 14 21 15.3457 21 17C21 18.6543 22.3457 20 24 20ZM24 16C24.5518 16 25 16.4487 25 17C25 17.5513 24.5518 18 24 18C23.4487 18 23 17.5513 23 17C23 16.4487 23.4487 16 24 16ZM25.8235 22H21.0588V24H23.7059V32H20V34H29V32H25.8235V22Z"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;

exports[`Icon examples Icon with custom colors should render correctly 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="v-icon v-icon-visa v-icon-high v-icon-home"
    focusable="false"
    height="48"
    style="--v-icon-primary: teal; --v-icon-secondary: orange;"
    viewBox="0 0 48 48"
    width="48"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      class="v-icon-primary"
      clip-rule="evenodd"
      d="M23.505 2.81648L45.6022 25L47 23.5967L23.505 0L13.5914 9.95223V4.38893H5.66062V17.914L0 23.5967L1.39781 25L7.64332 18.74V6.37938H11.6087V14.7592L23.505 2.81648ZM22.9 47L20.9 46.99L20.99 29.9989H14V46.99H12V28H23L22.9 47ZM26 28V39H37V28H26ZM35 32.5H32.5V30H35V32.5ZM30.5 30V32.5H28V30H30.5ZM28 34.5H30.5V37H28V34.5ZM32.5 37V34.5H35V37H32.5Z"
      fill-rule="evenodd"
    />
    <path
      class="v-icon-secondary"
      d="M44 47H25V45.0198H42V26.6238L24 8.80198L6 26.6238V45.0198H10V47H4V25.802L24 6L44 25.802V47Z"
    />
  </svg>
</div>
`;

exports[`Icon examples Icon with right-to-left support should render correctly 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="v-icon v-icon-visa v-icon-low v-icon-chevron-right v-icon-rtl"
    focusable="false"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      class="v-icon-primary"
      clip-rule="evenodd"
      d="M16.3402 12.0005L5 21.6717L6.5581 23.0005L18.6773 12.6649C19.1076 12.298 19.1076 11.703 18.6773 11.3361L6.5581 1.00049L5 2.32929L16.3402 12.0005Z"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;

exports[`Icon examples Low resolution should render correctly 1`] = `
<div>
  <svg
    aria-hidden="false"
    aria-label="info"
    class="v-icon v-icon-visa v-icon-low v-icon-information"
    focusable="false"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      class="v-icon-secondary"
      d="M12 24C5.38 24 0 18.62 0 12C0 5.38 5.38 0 12 0C18.62 0 24 5.38 24 12C24 18.62 18.62 24 12 24ZM12 2C6.49 2 2 6.49 2 12C2 17.51 6.49 22 12 22C17.51 22 22 17.51 22 12C22 6.49 17.51 2 12 2Z"
    />
    <path
      class="v-icon-primary"
      d="M15 19H9V17.0351H11.451V12.9874H9.43137V11.0225H13.4118V17.0351H15V19ZM12 9.53895C10.7549 9.53895 9.73529 8.51719 9.73529 7.26947C9.73529 6.02175 10.7549 5 12 5C13.2549 5 14.2647 6.02175 14.2647 7.26947C14.2647 8.51719 13.2549 9.53895 12 9.53895ZM12 6.96491C11.8333 6.96491 11.6961 7.10246 11.6961 7.26947C11.6961 7.43649 11.8333 7.57403 12 7.57403C12.1667 7.57403 12.3039 7.43649 12.3039 7.26947C12.3039 7.10246 12.1667 6.96491 12 6.96491Z"
    />
    <defs>
      <clippath>
        <rect
          height="24"
          width="24"
        />
      </clippath>
    </defs>
  </svg>
</div>
`;

exports[`Icon examples Tiny resolution should render correctly 1`] = `
<div>
  <svg
    aria-hidden="false"
    aria-label="info"
    class="v-icon v-icon-visa v-icon-tiny v-icon-information"
    focusable="false"
    height="16"
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      class="v-icon-primary"
      clip-rule="evenodd"
      d="M8 16C12.4183 16 16 12.4183 16 8C16 3.58172 12.4183 0 8 0C3.58172 0 0 3.58172 0 8C0 12.4183 3.58172 16 8 16ZM7 5V3H9V5H7ZM6 9H7V10V11H6V13H7H9H10V11H9V10V8V7H6V9Z"
      fill-rule="evenodd"
    />
    <defs>
      <clippath>
        <rect
          height="16"
          width="16"
        />
      </clippath>
    </defs>
  </svg>
</div>
`;

exports[`Icon examples Visa should render correctly 1`] = `
<div>
  <svg
    aria-hidden="false"
    aria-label="info"
    class="v-icon v-icon-visa v-icon-low v-icon-information"
    focusable="false"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      class="v-icon-secondary"
      d="M12 24C5.38 24 0 18.62 0 12C0 5.38 5.38 0 12 0C18.62 0 24 5.38 24 12C24 18.62 18.62 24 12 24ZM12 2C6.49 2 2 6.49 2 12C2 17.51 6.49 22 12 22C17.51 22 22 17.51 22 12C22 6.49 17.51 2 12 2Z"
    />
    <path
      class="v-icon-primary"
      d="M15 19H9V17.0351H11.451V12.9874H9.43137V11.0225H13.4118V17.0351H15V19ZM12 9.53895C10.7549 9.53895 9.73529 8.51719 9.73529 7.26947C9.73529 6.02175 10.7549 5 12 5C13.2549 5 14.2647 6.02175 14.2647 7.26947C14.2647 8.51719 13.2549 9.53895 12 9.53895ZM12 6.96491C11.8333 6.96491 11.6961 7.10246 11.6961 7.26947C11.6961 7.43649 11.8333 7.57403 12 7.57403C12.1667 7.57403 12.3039 7.43649 12.3039 7.26947C12.3039 7.10246 12.1667 6.96491 12 6.96491Z"
    />
    <defs>
      <clippath>
        <rect
          height="24"
          width="24"
        />
      </clippath>
    </defs>
  </svg>
</div>
`;
